### PR TITLE
nucleotide-count: use findWithDefault to accept sparse maps

### PR DESCRIPTION
**Pull request by sshine**
Bump version from 1.3.0.7 to 1.3.0.8.

@jitwit: I accidentally force-pushed `upstream/master` onto `jitwit/master` which was the basis for your feature branch, and for some reason, this triggered the PR to close. I've re-created the PR with your change attributed to you.

### CODE REVIEW

File: exercises/nucleotide-count/package.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: nucleotide-count
-version: 1.3.0.7
+version: 1.3.0.8
```
Comment (sshine) [inline exercises/nucleotide-count/package.yaml:2]: Bumping the version was the only thing I would have asked you to add.

File: exercises/nucleotide-count/test/Tests.hs [hunk 2]
```diff
@@ -13,7 +14,9 @@ main = hspecWith defaultConfig {configFastFail = True} specs
 specs :: Spec
 specs = do
 
-          let x `matchesMap` y = x `shouldBe` (Right . fromList) y
+          let x `matchesMap` y = shouldSatisfy x $ \case
+                Right count -> and [ findWithDefault 0 n count == c | (n,c) <- y ]
+                Left _ -> False
```
Comment (sshine) [inline exercises/nucleotide-count/test/Tests.hs:19]: I don't recall where I have this idea from, but I generally avoid `and` in favor of `all` (I think HLint said this), and I generally avoid list comprehensions when I can use a single higher-order function. In this particular case the list comprehension actually looks kind of nice and helps unpacking. Feel free to make it the way you like.

```suggestion
          let x `matchesMap` y = shouldSatisfy x $ \case
                Right count -> all (\(n,c) -> findWithDefault 0 n count == c) y
                Left _ -> False
```

Alternatively, I might have written it like:

```suggestion
          let count m (nuc, c) = findWithDefault 0 nuc m == c
              matchesMap m1 m2 = shouldSatisfy m1 $
                either (const False) (\m -> all (count m) m2)
```

**Review by sshine — COMMENTED** [review]
Excellent work!

This appears to the least invasive way to accept sparse maps as valid.

I'm sorry for the waiting time and the confusing wrt. opening and closing issues.

**Comment by sshine** [comment]
Okay, I should clearly take a break from the computer.

This PR was meant to go to `exercism/haskell` and not `jitwit/haskell`. :-D

**Comment by jitwit** [comment]
haha! yeah no worries, i'll respond to the comments over there

**Pull request closed without merging**